saleor/graphql/product/mutations/product_variant/product_variant_update.py

```diff
@@ -139,42 +138,38 @@ def set_track_inventory(cls, _info, instance, cleaned_input):
             instance.track_inventory = track_inventory
 
     @classmethod
-    def post_save_action(cls, info: ResolveInfo, instance, cleaned_input):
-        super().post_save_action(info, instance, cleaned_input)
-        channel_ids = ProductChannelListing.objects.filter(
-            product_id=instance.product_id
-        ).values_list("channel_id", flat=True)
-        cls.call_event(mark_active_catalogue_promotion_rules_as_dirty, channel_ids)
+    def _save_variant_instance(cls, instance, changed_fields):
+        update_fields = ["updated_at"] + changed_fields
+        instance.save(update_fields=update_fields)
 
     @classmethod
-    def _save(cls, info: ResolveInfo, instance, cleaned_input, base_fields_changed):
-        new_variant = instance.pk is None
-        variant_modified = False
+    def _save(cls, info: ResolveInfo, instance, cleaned_input, changed_fields) -> bool:
+        refresh_product_search_index = False
         with traced_atomic_transaction():
-            if base_fields_changed:
-                instance.save()
-                variant_modified = True
+            if changed_fields:
+                cls._save_variant_instance(instance, changed_fields)
+                if "sku" in changed_fields or "name" in changed_fields:
+                    refresh_product_search_index = True
             if stocks := cleaned_input.get("stocks"):
                 cls.create_variant_stocks(instance, stocks)
-                variant_modified = True
             if attributes := cleaned_input.get("attributes"):
                 AttributeAssignmentMixin.save(instance, attributes)
-                variant_modified = True
+                refresh_product_search_index = True
 
-            if variant_modified:
+            if refresh_product_search_index:
                 product_update_fields = ["updated_at", "search_index_dirty"]
                 instance.product.search_index_dirty = True
                 if not instance.product.default_variant:
                     instance.product.default_variant = instance
                     product_update_fields.append("default_variant")
```

Comment:
IMO we should set the default variant even if we don't need to refresh the search index.